app/test/non_determinism_test.go

```diff
@@ -0,0 +1,132 @@
+	"github.com/celestiaorg/celestia-app/v2/pkg/appconsts"
+	testutil "github.com/celestiaorg/celestia-app/v2/test/util"
+	"github.com/celestiaorg/celestia-app/v2/test/util/blobfactory"
+	"github.com/celestiaorg/celestia-app/v2/test/util/testfactory"
+	"github.com/celestiaorg/go-square/blob"
+	appns "github.com/celestiaorg/go-square/namespace"
+	"github.com/cosmos/cosmos-sdk/codec"
+	hd "github.com/cosmos/cosmos-sdk/crypto/hd"
+	keyring "github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// TestNonDeterminismBetweenMainAndV1 executes a set of different transactions,
+// produces an app hash and compares it with the app hash produced by v1.x
+func TestNonDeterminismBetweenMainAndV1(t *testing.T) {
+	const (
+		numBlobTxs, numNormalTxs = 5, 5
+	)
+
+	expectedAppHash := []byte{100, 237, 125, 126, 116, 10, 189, 82, 156, 116, 176, 136, 169, 92, 185, 12, 72, 134, 254, 175, 234, 13, 159, 90, 139, 192, 190, 248, 67, 9, 32, 217}
+
+	// Initialize testApp
+	testApp := testutil.NewTestApp()
+
+	enc := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	// Create deterministic keys
+	kr, pubKeys := DeterministicKeyRing(enc.Codec)
+
+	recs, err := kr.List()
+	require.NoError(t, err)
+	addresses := make([]string, 0, len(recs))
+
+	// Get the name of the records
+	for _, rec := range recs {
+		addresses = append(addresses, rec.Name)
+	}
+
+	// Apply genesis state to the app
+	_, _, err = testutil.ApplyGenesisState(testApp, pubKeys, 1_000_000_000, app.DefaultInitialConsensusParams())
+	require.NoError(t, err)
+
+	accinfos := queryAccountInfo(testApp, addresses, kr)
+
+	// Create a set of 10 deterministic sdk transactions
+	normalTxs := testutil.SendTxsWithAccounts(
+		t,
+		testApp,
+		enc.TxConfig,
+		kr,
+		1000,
+		addresses[0],
+		addresses[:numNormalTxs],
+		testutil.ChainID,
+	)
+
+	// Create a set of 5 deterministic blob transactions
+	blobTxs := blobfactory.ManyMultiBlobTx(t, enc.TxConfig, kr, testutil.ChainID, addresses[numBlobTxs+1:], accinfos[numBlobTxs+1:], testfactory.Repeat([]*blob.Blob{
+		blob.New(DeterministicNamespace(), []byte{1}, appconsts.DefaultShareVersion),
+	}, numBlobTxs), app.DefaultInitialConsensusParams().Version.AppVersion)
+
+	// Deliver sdk txs
+	for _, tx := range normalTxs {
+		resp := testApp.DeliverTx(abci.RequestDeliverTx{Tx: tx})
+		require.EqualValues(t, 0, resp.Code, resp.Log)
+	}
+
+	// Deliver blob txs
+	for _, tx := range blobTxs {
+		blobTx, ok := blob.UnmarshalBlobTx(tx)
+		require.True(t, ok)
+		resp := testApp.DeliverTx(abci.RequestDeliverTx{Tx: blobTx.Tx})
+		require.EqualValues(t, 0, resp.Code, resp.Log)
+	}
+
+	// Commit the state
+	testApp.Commit()
+
+	// Get the app hash
+	appHash := testApp.LastCommitID().Hash
+
+	// Assert that the app hash is equal to the app hash produced by v1.x
+	require.Equal(t, expectedAppHash, appHash)
```

Comment:
```suggestion
	assert.Equal(t, expectedAppHash, appHash)
```

Reply:
Thanks, updating the comment so that they both say assert or require would be nice. Since there are no more assertions after this line an assert will behave identically to a require.